### PR TITLE
Add auto push after commit merge

### PR DESCRIPTION
**Pull request by RissyRan**
# Description

Add auto push feature after the merge via cloudbuild

# Tests

Please describe the tests that you ran on Cloud VM to verify changes.

**Instruction and/or command lines to reproduce your tests:** ...
Set up cloudbuild in the project, and manually trigger the build

**List links for your tests (use go/shortn-gen for any internal link):** ...
Build [link](http://shortn/_kQC4V2b9bM)

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: pipeline/auto-push.cloudbuild.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+steps:
+  - name: gcr.io/cloud-builders/gsutil
```
Comment (will-cromar) [inline pipeline/auto-push.cloudbuild.yml:2]: nit: use `google/cloud-sdk` instead -- it gets updated more consistently https://registry.hub.docker.com/r/google/cloud-sdk

File: pipeline/auto-push.cloudbuild.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    args:
+      - '-c'
```
Comment (will-cromar) [inline pipeline/auto-push.cloudbuild.yml:4]: Should this just be

```
- scripts/upload-tests.sh
- gs://...
```

since `bash is already the entrypoint?

**Review by will-cromar — APPROVED** [review]
Nice!

File: pipeline/auto-push.cloudbuild.yml (hunk 1 above)
Comment (RissyRan) [reply pipeline/auto-push.cloudbuild.yml:2]: Sounds good. I was thinking to use a small image. As we are talking about 1min vs. 16s, it should be fine. Will update.

https://screenshot.googleplex.com/qqDZixy5n6WWuR4

File: pipeline/auto-push.cloudbuild.yml (hunk 2 above)
Comment (RissyRan) [reply pipeline/auto-push.cloudbuild.yml:4]: SG. Will update.

**Pull request merged**